core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -610,10 +614,63 @@ void releaseFetchLock() {
     }
 
     private void initialize() {
-        // Initialize the partition.
         log.debug("Initializing share partition: {}-{}", groupId, topicIdPartition);
+        // Initialize the partition by issuing an initialize RPC call to persister.
+        ReadShareGroupStateResult response;
+        try {
+            response = persister.readState(new ReadShareGroupStateParameters.Builder()
+                .setGroupTopicPartitionData(new GroupTopicPartitionData.Builder<PartitionIdLeaderEpochData>()
+                    .setGroupId(this.groupId)
+                    .setTopicsData(Collections.singletonList(new TopicData<>(topicIdPartition.topicId(),
+                        Collections.singletonList(PartitionFactory.newPartitionIdLeaderEpochData(topicIdPartition.partition(), 0)))))
```

Comment:
You should use the current leader epoch rather than 0 here.

Reply:
I have added following task for myself to fix it: https://issues.apache.org/jira/browse/KAFKA-17002